File: sdk/arena-python-sdk/samples/pytorchjob_test.py
#!/usr/bin/env python
import os
import sys
import time
from arenasdk.client.client import ArenaClient
from arenasdk.enums.types import *
from arenasdk.exceptions.arena_exception import *
from arenasdk.training.pytorch_job_builder import *
from arenasdk.logger.logger import LoggerBuilder

def main():
    print("start to test arena-python-sdk")
    client = ArenaClient("","default","info","arena-system")
    print("create ArenaClient succeed.")
    print("start to create pytorchjob")
    job_name = "pytorchjob-test"
    job_type = TrainingJobType.PytorchTrainingJob
    try:
        # build the training job 
        job =  PytorchJobBuilder().with_name(job_name).with_workers(1).with_gpus(1).with_sync_mode("git").with_sync_source("https://github.com/happy2048/mnist-pytorch.git").with_image("registry.cn-beijing.aliyuncs.com/ai-samples/pytorch-with-tensorboard:1.5.1-cuda10.1-cudnn7-runtime").with_command("'python /root/code/mnist-pytorch/mnist.py --backend gloo'").build()
        # if training job is not existed,create it 
        if not client.training().get(job_name,job_type):
            output = client.training().submit(job)
            print(output)
        else:
            print("the job {} has been created,skip to create it".format(job_name))
        count = 0
        # waiting training job to be running 
        while True:
            if count > 160:
                raise Exception("timeout for waiting job to be running")
            jobInfo = client.training().get(job_name,job_type)
            if jobInfo.get_status() == TrainingJobStatus.TrainingJobPending:
                print("job status is PENDING,waiting...")
                count = count + 1
                time.sleep(5)
                continue
            print("current status is {} of job {}".format(jobInfo.get_status().value,job_name))
            break
        # get the training job logs 
        logger = LoggerBuilder().with_accepter(sys.stdout).with_follow().with_since("5m")
        jobInfo.get_instances()[0].get_logs(logger)
        # display the training job information
        print(str(jobInfo))
        # delete the training job 
        client.training().delete(job_name,job_type)
    except ArenaException as e:
        print(e)

main()
    

File: sdk/arena-java-sdk/src/test/java/com/github/kubeflow/arena/PyTorchJobTest.java
package com.github.kubeflow.arena;

import com.github.kubeflow.arena.client.ArenaClient;
import com.github.kubeflow.arena.exceptions.ArenaException;
import com.github.kubeflow.arena.model.training.PytorchJobBuilder;
import com.github.kubeflow.arena.model.training.TrainingJob;
import org.junit.Test;

import java.io.IOException;
import java.util.HashMap;
import java.util.Map;

public class PyTorchJobTest {

    @Test
    public void testSubmit() throws ArenaException, IOException {
        ArenaClient client = new ArenaClient();

        Map<String, String> labels = new HashMap<>();
        labels.put("arena.kubeflow.org/console-user", "1234567890");

        String command = "python /root/code/mnist-pytorch/mnist.py --backend gloo";

        TrainingJob job = new PytorchJobBuilder()
                .ttlAfterFinished("30s")
                .name("test-pytorchjob-1")
                .gpus(1)
                .labels(labels)
                .syncMode("git")
                .syncSource("https://code.aliyun.com/370272561/mnist-pytorch.git")
                .logDir("/training_logs")
                .image("registry.cn-beijing.aliyuncs.com/ai-samples/pytorch-with-tensorboard:1.5.1-cuda10.1-cudnn7-runtime")
                .imagePullPolicy("IfNotPresent")
                .command(command)
                .build();

        String result = client.training().namespace("default-group").submit(job);
        System.out.println(result);
    }

}
